ns/nslogic.go

```diff
@@ -610,6 +623,43 @@ func WatchingNamespace(clientset *kubernetes.Clientset, requestedNS, installatio
 	return false
 }
 
+// CreatePGODefaultServiceAccount creates the default SA
+func CreatePGODefaultServiceAccount(clientset *kubernetes.Clientset, targetNamespace string) error {
+
+	_, found, _ := kubeapi.GetServiceAccount(clientset, PGO_DEFAULT_SERVICE_ACCOUNT, targetNamespace)
+	if found {
+		log.Infof("serviceaccount %s already exists, will delete and re-create", PGO_DEFAULT_SERVICE_ACCOUNT)
+		err := kubeapi.DeleteServiceAccount(clientset, PGO_DEFAULT_SERVICE_ACCOUNT, targetNamespace)
+		if err != nil {
+			log.Errorf("error deleting serviceaccount %s %s", PGO_DEFAULT_SERVICE_ACCOUNT, err.Error())
+			return err
+		}
+	}
+	var buffer bytes.Buffer
+	err := config.PgoDefaultServiceAccountTemplate.Execute(&buffer,
+		PgoDefaultServiceAccount{
+			TargetNamespace: targetNamespace,
+		})
+	if err != nil {
+		log.Error(err.Error() + " on " + config.PGODefaultServiceAccountPath)
+		return err
+	}
+	log.Info(buffer.String())
+
+	rb := v1.ServiceAccount{}
+	err = json.Unmarshal(buffer.Bytes(), &rb)
+	if err != nil {
+		log.Error("error unmarshalling " + config.PGODefaultServiceAccountPath + " json ServiceAccount " + err.Error())
+		return err
+	}
+
+	err = kubeapi.CreateServiceAccount(clientset, &rb, targetNamespace)
+	if err != nil {
+		return err
+	}
+	return err
```

Comment:
Did you intend to log the err or do something else with it here? Otherwise, why not just return `err` or even just simply `return kubeapi.CreateServiceAccount(...)`?

Reply:
Oof. I copy-pasta'd from the other SA create functions. I'll amend this and the others.